api4/user_test.go

```diff
@@ -6164,27 +6261,27 @@ func TestMaintainUnreadMentionsInThread(t *testing.T) {
 
 	// test self mention, shouldn't increase mention count
 	postAndCheck(t, client, &model.Post{ChannelId: th.BasicChannel.Id, Message: "testReply @" + th.BasicUser.Username, RootId: rpost.Id})
-	// count shouldn't increase
-	checkThreadList(th.Client, th.BasicUser.Id, 1, 1)
+	// mention should be 0 after self reply
+	checkThreadList(th.Client, th.BasicUser.Id, 0, 1)
 
 	// test DM
 	dm := th.CreateDmChannel(th.SystemAdminUser)
 	dm_root_post, _ := postAndCheck(t, client, &model.Post{ChannelId: dm.Id, Message: "hi @" + th.SystemAdminUser.Username})
 
 	// no changes
-	checkThreadList(th.Client, th.BasicUser.Id, 1, 1)
+	checkThreadList(th.Client, th.BasicUser.Id, 0, 1)
 
-	// post reply by the same user
+	// post reply by the BasicUser
```

Comment:
If it's a different user, shouldn't the count increase? Same for below as well.

Reply:
no sorry, `BasicUser` is the same user. I'll add that to the comment